app/models/adviser.rb

```diff
@@ -46,6 +48,10 @@ def field_order
 
   private
 
+  def upcase_postcode
```

Comment:
In order to achieve the same result, I would personally override the setter method, i.e.:

```ruby
def postcode=(postcode)
  write_attribute(:postcode, postcode.upcase)
end
```

In current implementation, postcode will remain as it was provided until firing the validation. In this specific case it probably won't cause any problems, but there're scenarios where it might cause problems (eg. two-step save, where you actually validate the existence of given address against an external service, before dumping the record into the db).

Of course, my note doesn't stop me from :+1:. Just consider if the code is sufficiently future-proof.